### PR TITLE
Eco 156/add bi events

### PR DESCRIPTION
**Pull request by OrenZak**
- Changed playground server URL to https
- Add bi events
- Updated tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,4 +51,11 @@ google-services.json
...
 freeline/
 freeline_project_description.json
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+# bi events
+bi-events-generator/node_modules
+bi-events-generator/*.js
+bi-events-generator/*.js.map
```
Comment (YohayBar) [inline .gitignore]: no need

File: kin-ecosystem-core/build.gradle [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+    defaultConfig {
+        minSdkVersion 16
+        targetSdkVersion 26
+        versionCode 1
+        versionName "${KIN_ECOSYSTEM_CORE_VERSION_NAME}"
+
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles 'proguard-rules.pro'
```
Comment (YohayBar) [inline kin-ecosystem-core/build.gradle]: no need for now

File: kin-ecosystem-core/build.gradle [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
```
Comment (YohayBar) [inline kin-ecosystem-core/build.gradle]: ?

File: kin-ecosystem-core/src/main/java/kin/ecosystem/core/network/ApiClient.java [hunk 4]
```diff
@@ -64,7 +62,7 @@
 
 public class ApiClient {
 
-    private String basePath = Configuration.getEnvironment().getEcosystemServerUrl();
+    private String basePath;
 //    private String basePath = "http://10.0.2.2:3000/v1";
```
Comment (YohayBar) [inline kin-ecosystem-core/src/main/java/kin/ecosystem/core/network/ApiClient.java]: remove

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/EventCommonDataUtil.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+					@Override
+					public String get() {
+						return AuthRepository.getInstance().getDeviceID();
+					}
+				});
+				commonProxy.setCarrier(carrierName);
+				commonProxy.setOs(VERSION.RELEASE);
+				commonProxy.setDeviceManufacturer(Build.MANUFACTURER);
+				commonProxy.setDeviceModel(Build.MODEL);
+				commonProxy.setLanguage(Locale.getDefault().getDisplayLanguage());
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/EventCommonDataUtil.java]: DynamicValue - can change

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/Kin.java [hunk 6]
```diff
@@ -5,10 +5,25 @@
...
+import com.kin.ecosystem.bi.EventLogger;
+import com.kin.ecosystem.bi.EventLoggerImpl;
+import com.kin.ecosystem.bi.EventsStore;
+import com.kin.ecosystem.bi.EventsStore.ClientModifier;
+import com.kin.ecosystem.bi.EventsStore.CommonModifier;
+import com.kin.ecosystem.bi.EventsStore.DynamicValue;
+import com.kin.ecosystem.bi.EventsStore.UserModifier;
+import com.kin.ecosystem.bi.events.ClientProxy;
+import com.kin.ecosystem.bi.events.CommonProxy;
+import com.kin.ecosystem.bi.events.EntrypointButtonTapped;
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/Kin.java]: imports?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/bi/EventLoggerImpl.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            }
+        }
+        return  instance;
+    }
+
+
+    @Override
+    public void send(Event event) {
+        try {
+            event.getCommon().setEventId(getUniqueID());
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/bi/EventLoggerImpl.java]: ? remove used via dynamic

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/bi/EventLoggerImpl.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                public void onDownloadProgress(long bytesRead, long contentLength, boolean done) {
+
+                }
+            });
+        } catch (ApiException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private UUID getUniqueID() {
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/bi/EventLoggerImpl.java]: also remove

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/bi/EventsApi.java [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+		String localVarPath = "/eco_";
+
+		List<Pair> localVarQueryParams = new ArrayList<Pair>();
+		List<Pair> localVarCollectionQueryParams = new ArrayList<Pair>();
+
+		Map<String, String> localVarHeaderParams = new HashMap<String, String>();
+
+		final String eventUUID = event.getCommon().getEventId().toString();
+		localVarHeaderParams.put("X-REQUEST-ID", eventUUID);
+		final String eventId = apiClient.parameterToString(eventUUID);
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/bi/EventsApi.java]: ?

File: kin-ecosystem-sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java [hunk 10]
```diff
@@ -52,6 +65,7 @@ public void tearDown() throws Exception {
     @Test
     public void getStartedClicked_AnimationEndedNavigateMP() throws Exception {
         splashPresenter.getStartedClicked();
+        verify(eventLogger).send(any(WelcomeScreenButtonTapped.class));
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/test/java/com/kin/ecosystem/splash/presenter/SplashPresenterTest.java]: amazing :)

File: kin-ecosystem-sdk/src/test/java/com/kin/ecosystem/data/order/OrderRepositoryTest.java [hunk 11]
```diff
@@ -202,10 +220,111 @@ public void onChanged(Order value) {
         assertNull(orderRepository.getOpenOrder().getValue());
         verify(offerRepository).setPendingOfferByID(null);
 
+        verify(eventLogger, never()).send(any(SpendOrderCompleted.class));
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/test/java/com/kin/ecosystem/data/order/OrderRepositoryTest.java]: why do we need this "never" test?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/OrderRepository.java [hunk 12]
```diff
@@ -160,6 +174,12 @@ public void onChanged(Payment payment) {
 		}
 	}
 
+	private void sendEarnPaymentConfirmed(Payment payment) {
+		if (payment.isSucceed() && payment.getAmount() != null && payment.isEarn()) {
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/OrderRepository.java]: do you need to check payment != null?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/history/presenter/CouponDialogPresenter.java [hunk 13]
```diff
@@ -3,26 +3,37 @@
...
+        this.order = order;
+        this.redeemTrigger = redeemTrigger;
     }
 
     @Override
     public void onAttach(ICouponDialog view) {
         super.onAttach(view);
-        loadInfo();
+		loadInfo();
+		eventLogger.send(SpendRedeemPageViewed.create(redeemTrigger, (double) order.getAmount(), order.getOfferId(), order.getOrderId()));
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/history/presenter/CouponDialogPresenter.java]: check order!= null?

File: kin-ecosystem-sdk/src/test/java/com/kin/ecosystem/data/order/OrderRepositoryTest.java (hunk 11 above)
Comment (OrenZak) [reply kin-ecosystem-sdk/src/test/java/com/kin/ecosystem/data/order/OrderRepositoryTest.java]: i want to check that no spend even it called, this is earn submission.

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/OrderRepository.java (hunk 12 above)
Comment (OrenZak) [reply kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/order/OrderRepository.java]: no, payment never null when it gets here

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/history/presenter/CouponDialogPresenter.java (hunk 13 above)
Comment (OrenZak) [reply kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/history/presenter/CouponDialogPresenter.java]: order marked on the constructor as nonnull

**Pull request merged**